spec/railtie_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe ApplicationController, type: :controller do
+  render_views
+
+  controller do
+    around_action :test_arel_toolkit
+
+    class IgnoreFirstEntry
+      def self.call(arel, _context)
+        arel.skip(1)
+      end
+    end
+
+    def test_arel_toolkit(&block)
+      Arel.middleware.apply([IgnoreFirstEntry]) do
+        yield block
+      end
+    end
+
+    def index
+      render plain: User.all.map(&:username).to_json
+    end
+  end
+
+  describe 'index' do
+    it 'returns a sql generated by the arel toolkit' do
+      User.create! username: 'Maarten'
+      User.create! username: 'Willian'
+
+      puts User.all.map &:username
+
+      get :index
+
+      expect(response.body).to eq ['Willian'].to_json
```

Comment:
All this work for only this assertion :)

Reply:
😍